### PR TITLE
Dfs dart

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2172 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Depth first search is implemented using adjacency matrix in Dart
- Works well for both directed and undirected graphs.

#### Languages Used:
- Dart

#### Files Added:
- Depth_First_Search.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Rukmini-Meda** [comment]
Fixes #2172

File: Breadth_First_Search/Breadth_First_Search.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Required library to use inbuilt queues in Dart
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.dart]: Please remove BFS code from this commit

File: Depth_First_Search/Depth_First_Search.dart [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// Required library to use inbuilt queue in LIFO manner in Dart
+import 'dart:collection';
+// Required library to use standard output facility of dart
+import 'dart:io';
+
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.dart]: Add space around operators throughout the file

File: Depth_First_Search/Depth_First_Search.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                queue.add(i);
+                visited[i]=1;
+            }
+        }
+    } 
+}
+
+void main(){
+
+    int no_vertices=5;
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.dart]: Please take dynamic inputs wherever applicable

File: Depth_First_Search/Depth_First_Search.dart [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    print("Vertices in Depth First Search of the graph in sample input is: ");
+
+    // Depth First Search of the above graph
+    Depth_First_Search(graph,source,no_vertices);
+
+    // Sample Output
+    // Vertices in Depth First Search of the graph in sample input is:
+    // 0 3 4 2 1
+
+}
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.dart]: Add a blank line at the end of the file

File: Breadth_First_Search/Breadth_First_Search.dart (hunk 1 above)
Comment (Rukmini-Meda) [reply Breadth_First_Search/Breadth_First_Search.dart]: Changes made

File: Depth_First_Search/Depth_First_Search.dart (hunk 2 above)
Comment (Rukmini-Meda) [reply Depth_First_Search/Depth_First_Search.dart]: Changes made

File: Depth_First_Search/Depth_First_Search.dart (hunk 3 above)
Comment (Rukmini-Meda) [reply Depth_First_Search/Depth_First_Search.dart]: Made changes to take input dynamically

File: Depth_First_Search/Depth_First_Search.dart (hunk 4 above)
Comment (Rukmini-Meda) [reply Depth_First_Search/Depth_First_Search.dart]: Added blank line

**Comment by Rukmini-Meda** [comment]
I have made all the required changes. Please let me know if it still requires any further changes. Thank you.

File: Depth_First_Search/Depth_First_Search.dart [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    print("Enter number of edges:");
+    var input1 = stdin.readLineSync();
+    int no_edges = int.parse(input);
+    print("Enter source and destination vertices of edges:\n");
+    print("Note: Enter source and destination vertices as 0,1,2,...");
+    
+    for(int i = 1; i <= no_edges; i++){
+      
+      print("Edge #$i:");
+
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.dart]: Indentation inside for loop & if, else should be made to 4 space

**Comment by Rukmini-Meda** [comment]
I have changed the indentation. Please review and reply.

**Pull request merged**